use_cases/eluc/predictors/neural_network/neural_net_predictor.py

```diff
@@ -0,0 +1,297 @@
+        if not load_path.exists():
+            raise FileNotFoundError(f"Path {path} does not exist.")
+        
+        # Initialize model with config
+        with open(load_path / "config.json", "r", encoding="utf-8") as f:
+            config = json.load(f)
+        self.__init__(**config)
+
+        self.model = ELUCNeuralNet(len(self.features), self.hidden_sizes, self.linear_skip, self.dropout)
+        self.model.load_state_dict(torch.load(load_path / "model.pt"))
+        self.model.to(self.device)
+        self.model.eval()
+        self.scaler = joblib.load(load_path / "scaler.joblib")
+
+
+    def save(self, path: str):
+        """
+        Saves model, config, and scaler into format for loading.
+        Generates path to folder if it does not exist.
+        :param path: path to folder to save model files.
+        """
+        if self.model is None:
+            raise ValueError("Model not fitted yet.")
+        save_path = Path(path)
+        save_path.mkdir(parents=True, exist_ok=True)
+
+        config = {
+            "features": self.features,
+            "label": self.label,
+            "hidden_sizes": self.hidden_sizes,
+            "linear_skip": self.linear_skip,
+            "dropout": self.dropout,
+            "device": self.device,
+            "epochs": self.epochs,
+            "batch_size": self.batch_size,
+            "optim_params": self.optim_params,
+            "train_pct": self.train_pct,
+            "step_lr_params": self.step_lr_params
+        }
+        with open(save_path / "config.json", "w", encoding="utf-8") as f:
+            json.dump(config, f)
+        torch.save(self.model.state_dict(), save_path / "model.pt")
+        joblib.dump(self.scaler, save_path / "scaler.joblib")
+
+
+    def fit(self, X_train: pd.DataFrame, y_train: pd.Series, X_val=None, y_val=None, X_test=None, y_test=None, log_path=None, verbose=False) -> dict:
+        """
+        Fits neural network to given data using predefined parameters and hyperparameters.
+        If no features were specified we use all the columns in X_train.
+        We scale based on the training data and apply it to validation and test data.
+        AdamW optimizer is used with L1 loss.
+        :param X_train: training data, may be unscaled and have excess features.
+        :param y_train: training labels.
+        :param X_val: validation data, may be unscaled and have excess features.
+        :param y_val: validation labels.
+        :param X_test: test data, may be unscaled and have excess features.
+        :param y_test: test labels.
+        :param log_path: path to log training data to tensorboard.
+        :param verbose: whether to print progress bars.
+        :return: dictionary of results from training containing time taken, best epoch, best loss, and test loss if applicable.
+        """
+        if not self.features:
+            self.features = X_train.columns.tolist()
+        self.label = y_train.name
```

Comment:
Features and label are set here during fitting. If we haven't specified a set of features to use we just take them all from X_train. The label is taken from y_train.